File: 백준/Bronze/17027. Shell Game/Shell Game.py
import sys
input = lambda: sys.stdin.readline().rstrip()


def main():
    N = int(input())
    lst = [list(map(int, input().split())) for _ in range(N)]
    dap = 0
    for i in range(1, 4):
        tmp = 0
        for n in range(N):
            a, b, g = lst[n]
            if i == a:
                i = b
            elif i == b:
                i = a
            if i == g:
                tmp += 1
        if dap < tmp:
            dap = tmp
    print(dap)


if __name__ == '__main__':
    main()

File: 백준/Bronze/17027. Shell Game/Shell Game.c
#include <stdio.h>

int main() {
    int N;
    scanf("%d", &N);
    int lst[N][3];
    for (int i = 0; i < N; i++) {
        scanf("%d %d %d", &lst[i][0], &lst[i][1], &lst[i][2]);
    }
    int dap = 0;
    for (int i = 1; i <= 3; i++) { 
        int tmp = 0;
        int current = i;
        for (int n = 0; n < N; n++) {
            int a = lst[n][0];
            int b = lst[n][1];
            int g = lst[n][2];
            if (current == a) {
                current = b;
            } else if (current == b) {
                current = a;
            }
            if (current == g) {
                tmp++;
            }
        }
        if (dap < tmp) {
            dap = tmp;
        }
    }
    printf("%d\n", dap);
    return 0;
}